cmd/cli/kubectl-kyverno/commands/test/test.go

```diff
@@ -98,6 +98,51 @@ func runTest(out io.Writer, testCase test.TestCase, registryAccess bool, auditWa
 	} else {
 		fmt.Fprintln(out, "  Applying", len(policies)+len(validatingAdmissionPolicies), pluralize.Pluralize(len(policies)+len(validatingAdmissionPolicies), "policy", "policies"), "to", len(uniques), pluralize.Pluralize(len(uniques), "resource", "resources"), "...")
 	}
+	rulesmap := make(map[string]string)
+	policymap := make(map[string]string)
+	for _, res := range testCase.Test.Results {
+		if res.IsValidatingAdmissionPolicy {
+			continue
+		}
+		var ns string
+		var n bool
+		r := false
```

Comment:
I don't think we need these variables. If the policy name declared in the `results` doesn't match a policy from the given policy file, then store this name directly in `unknownPolicies`. Hence, there will be no need for the `n` variable. And do the same for rules.

Reply:
Ok will be update

Reply:
This variables are used to find the rule names if it is not found in the rules for loop.
So we can change the value to true if found else false.
In the last for if condition the map insertion is done based on the values?

Reply:
Variables `n` and `r` don't indicate their usage. You can rename them to `policyExists` and `ruleExists` or something similar.

In addition, `r` has an initial value of false whereas `n` has no initial values.